avocado/utils/process.py

```diff
@@ -199,22 +200,41 @@ def get_children_pids(parent_pid, recursive=False):
     return children
 
 
-def kill_process_tree(pid, sig=signal.SIGKILL, send_sigcont=True):
+def kill_process_tree(pid, sig=signal.SIGKILL, send_sigcont=True,
+                      wait=False):
     """
     Signal a process and all of its children.
 
     If the process does not exist -- return.
 
     :param pid: The pid of the process to signal.
     :param sig: The signal to send to the processes.
+    :param wait: How long to wait for the pid(s) to die (False=don't wait,
```

Comment:
My suggestion here is to use a single type for `wait`, instead of multiple types (None, False, integer).  IMO it's easier to understand that:
 * `0` = do not wait
 * `< 0` = wait indefinitely
 * `> 0` = wait the given amount of seconds

Also, maybe `timeout` as the name of the parameter is better aligned with the other parameters in this module that are similar in nature.

Reply:
sure, no strong opinion about this...